Pinta.Gui.Widgets/Widgets/Canvas/CanvasWindow.cs

```diff
@@ -110,6 +117,42 @@ public CanvasWindow (Document document)
 		};
 
 		AddController (motion_controller);
+
+		gesture_zoom = GestureZoom.New ();
+		gesture_zoom.SetPropagationPhase (PropagationPhase.Bubble);
+		gesture_zoom.OnScaleChanged += Gesture_zoomOnOnScaleChanged;
+		gesture_zoom.OnBegin += (_, _) => last_scale_delta = 0;
+		AddController (gesture_zoom);
+	}
+
+	private void Gesture_zoomOnOnScaleChanged (object? sender, EventArgs e)
+	{
+		// Allow the user to zoom in/out by pinching the trackpad
+		double pinchDelta = gesture_zoom.GetScaleDelta () - 1 - last_scale_delta;
+		if (pinchDelta < 0) {
+			if (cumulative_zoom_amount > 0) {
+				cumulative_zoom_amount = 0;
+			}
+			cumulative_zoom_amount += pinchDelta;
+			Debug.WriteLine(cumulative_zoom_amount + " ::: " + pinchDelta + " ::: " + last_scale_delta);
+			if (cumulative_zoom_amount <= -ZoomThresholdPinch) {
+				Debug.WriteLine("Zoom out");
```

Comment:
The debug prints can be removed once things are finalized

Reply:
I could've sworn I deleted all of those! Fixed.